### DISCUSSION TITLE
Vers. 0.4 beta is out! Major update for noScribe

### DISCUSSION THREAD

**Discussion by kaixxx** [kaixxx/noScribe#28] [Announcements]
Version 0.4 is a major update, so many things can go wrong...
- Please post your experiences here (positive or negative). 
- You can also open an issue to report an error. 

Thank you!

**Comment by tobias-d** [comment]
First of all: Thank you for the update and the software which has profoundly improved my working life.

Unfortunately, the recent Beta crashes after loading the whisper plugin. The speaker recognition works but than the app crashes completely. I am on Windows 11.

**Reply by kaixxx** [reply]
Thannk you for the report. I haven't heard about this problem yet. 
With "crashing" you mean that the app is not doing anything? Or does windows give you a warning message about the app not resonding anymore?
If you did not select a language (but have it still on "auto"), whisper may take a while to figure out the correct language. Could you retry with a preselected language (if your werde on auto before)?

Thanks

**Reply by tobias-d** [reply]
The app just completely shuts down without any error message. When I restart the app all settings are lost. 
I have only tried with pre-set language.

**Reply by kaixxx** [reply]
Strange. Can you send the logfile from the attempted transcript to kai.droege (ät) hslu.ch? I describe here (last paragraph) where to find this file: https://github.com/kaixxx/noScribe#advanced-options

**Reply by tobias-d** [reply]
The log file is empty but the file was created.

Just re-installed 0.3 and there is no issue.

Also tried different audio files. One has only 15min.

**Reply by Schmukas95** [reply]
I have the exact same Problem. Unfortunately my log file is completely empty

**Reply by kaixxx** [reply]
Ok, that's bad. Let me think about a way to narrow down the problem. I'll be back in an hour or so.

**Reply by amru39** [reply]
Facing the same issue. The embeddings is completed for speaker recognition and then the app just shuts down. Attaching log file. 
[KII6.log](https://github.com/kaixxx/noScribe/files/13056068/KII6.log). I just tried for the initial minute it worked and did not close.

**Reply by Schmukas95** [reply]
with the new version I could now also create a log file. The crash always happens at the same time. couldnt get trough the first minute 

[TestAudio.log](https://github.com/kaixxx/noScribe/files/13056325/TestAudio.log)

**Reply by kaixxx** [reply]
Ok, thank you. I can now see where exactly the error happens. But I am still not sure what the reason might be.
One idea: You seem to have quite powerful computers. Do you also have a NVIDIA graphics card? It may be that faster-whisper tries to use GPU-acceleration and fails. I cannot test this since I don't have a NVIDIA card. 
I will compile a new version that forces whisper to run only on the CPU.

**Reply by amru39** [reply]
Yes, a NVIDIA T1200 Laptop GPU. Thanks for working on fixing this. Really appreciate it.

**Reply by kaixxx** [reply]
> Yes, a NVIDIA T1200 Laptop GPU.

So this might in fact be the problem. Let's hope it is that simple. Here is a new exe, which now forces whisper to run only on the cpu: https://drive.switch.ch/index.php/s/SXCbPkMhc3BNnCg Place the file in your program folder as described below. 

It is a little sad to disable gpu-acceleration for now. But we can work on that later.

**Reply by amru39** [reply]
Maybe related to GPU-acceleration being disabled but the CPU usage is on the higher side (above 85% consistently). I manually limited Diarize.exe to five cores to bring it down to around 45%. Testing on a 30 minute file.

**Reply by Schmukas95** [reply]
thank you very much for your efforts, Kai! Your tool has already saved me many hours of work.

I also use an NVIDIA graphics card, GTX 1060

**Reply by kaixxx** [reply]
It seems that we have found the error. Can you confirm that it runs fine with the exe linked above? I'm about to update the installer.
Thanks!

**Reply by gernophil** [reply]
@kaixxx, how do you feel about an option (maybe an advanced option in the `.yaml`) to throttle CPU usage? I'm seeing this with macOS, too. If using `cpu` and not `mps` CPU usage is extraordinarily high.

**Reply by BabyFnord** [reply]
If possible, configuring cpu usage in the GUI is a good idea. As a user with no other option than slow/older machines, it is reasonable to have the vanilla setting use all available cpu power there is—why would you choose to throttle an already very slow process on a very slow machine, anyway?

**Reply by kaixxx** [reply]
Regarding CPU usage, @gernophil  and @BabyFnord: I would be happy to add such an option in the advanced-options. But I am not sure how to implement it. With faster-whisper, it is easy: there is an option to reduce the number of cores used. But for pyannote, I don't know how to do this. Any ideas?

**Reply by kaixxx** [reply]
@Schmukas95 @tobias-d @amru39 
I've released a new version 0.4.1 which should fix the issue. Please test and report back if you'll find the time, thanks!

This bug was really annoying me because it came in only a few hours after I made vers. 0.4 non-beta. Thank you for your help in narrowing the problem down!
I am thinking about reintroducing GPU-support in the future since it could bring down transcription times by a lot. Unfortunatly, I don't have a suitable graphics card (NVIDIA only). Would you help testing such a new version? It may take a few weeks until I get around working on this.

**Reply by amru39** [reply]
> It seems that we have found the error. Can you confirm that it runs fine with the exe linked above? I'm about to update the installer. Thanks!

Yes, it worked. Attaching log file for reference. 
[KII6.log](https://github.com/kaixxx/noScribe/files/13061159/KII6.log)

**Reply by amru39** [reply]
> @Schmukas95 @tobias-d @amru39 I've released a new version 0.4.1 which should fix the issue. Please test and report back if you'll find the time, thanks!
> 
> This bug was really annoying me because it came in only a few hours after I made vers. 0.4 non-beta. Thank you for your help in narrowing the problem down! I am thinking about reintroducing GPU-support in the future since it could bring down transcription times by a lot. Unfortunatly, I don't have a suitable graphics card (NVIDIA only). Would you help testing such a new version? It may take a few weeks until I get around working on this.

Testing it right now. Will report back later today/tomorrow. Yes, I can help to test any further versions. I have a couple of files that I need to transcribe.

**Reply by amru39** [reply]
Quick update. Worked perfectly. With no other background process/application running, the same recording took 15 minutes less to transcribe. The CPU usage during embedding was around 50 percent and for the actual transcription around 80 percent.

**Comment by kaixxx** [comment]
I have compiled a new version of noScribe. This is not going to fix the original problem, but should give us more information in the log files to narrow down the error. 
You can download the new noScribe.exe from here: https://drive.switch.ch/index.php/s/dmnjyNGI16XRQ4a This is not an installer, just the main application. 
You have to place the file inside the installation folder of noScribe 0.4, which should be "C:\Program Files (x86)\noScribe". Please replace the exe that is already there (you can also rename the old exe if you want to keep it).
If you now run noScribe.exe, a terminal window will appear which will give some information during the transcrption process. More importantly, the log file should now actually have some info in it. 
Could you post the logfile here? Please delete any sensitive information beforehand. 
Thanks!

**Reply by kaixxx** [reply]
One addition: You can test with a much shorter section than 15 minutes to make your life easier. This should not make a difference. Just set the "Stop" field to 00:01:00 for the first minute of your audio.

**Reply by tobias-d** [reply]
Okay, it is working now. Thanks a lot! I am also happy to test with my NVIDIA card, would be very curious how fast it can be with a RTX A4000 :)

**Reply by kaixxx** [reply]
@tobias-d Perfect, happy to hear that. I'll come back to you if I find the time to work on the cuda-integration.

**Comment by tobias-d** [comment]
Won’t be able to test it before Monday. But yes, also have a powerful NVIDIA Graphics Card.

**Reply by kaixxx** [reply]
Thank you. Let's see - I hope you'll have a working version by monday.

**Comment by BabyFnord** [comment]
On a side-note, the macOS version provided by @gernophil runs 💯 fine on my late-2012 MacBook Pro and macOS Mojave 10.14.6. You'll need to exchange two .dylibs (see attached .zip), though. To anyone inclined on using trailing edge Macs & macOS for noScribe, the following Terminal commands get the adaptation done on v0.4 beta (download and unpack that .zip first):
`cp $HOME/Downloads/libtiff.6.dylib /Applications/noScribe.app/Contents/Frameworks/PIL/__dot__dylibs/`
`cp $HOME/Downloads/libX11.6.dylib /Applications/noScribe.app/Contents/Frameworks/av/__dot__dylibs/`
[noScribe_x64_Libs.zip](https://github.com/kaixxx/noScribe/files/13059172/noScribe_x64_Libs.zip)

Thanks to @kaixxx and @gernophil for their patient beta-test support!

**Comment by kaixxx** [comment]
## Note: Vers. 0.4 had a problem in combination with NVIDIA graphics cards on Windows. Vers. 0.4.1 fixes this. If you did not experience this bug, you don't have to update.

**Comment by gernophil** [comment]
Just realeased 0.4.2 for macOS, which is using torchaudio 2.1.0. If you are using MPS (macOS GPU) then I strongly recommend to switch to that Version, since the old speaker recognition was really error prone.

**Comment by kaixxx** [comment]
@Schmukas95 @tobias-d @amru39 
A new version with cuda is out. Could you help testing it? See https://github.com/kaixxx/noScribe/discussions/50

**Reply by tobias-d** [reply]
Sorry for the late reply. I have been using the new version quite a lot recently and it works perfectly fine. Transcription times vary of course but it is very fast. 
Last one: 1hr recording with 3 speakers: 100min

I was wondering if the new version also has an effect on the quality of the transcription. I had the feeling it was more accurate but it is hard to tell because there are so many factors to consider.

**Reply by kaixxx** [reply]
Great, thank you. The transcription quality should be the same, but its hard to test really. However, there will be a new version released in the next couple of days with some small improvements. Watch out..
